diff --git a/src/plugins/sync-plugin.js b/src/plugins/sync-plugin.js
index f719448f24e841282f4bbd29cc5167f33fb06596..0d0ce3c694d59e889af5df6456cf4135c5b22c44 100644
--- a/src/plugins/sync-plugin.js
+++ b/src/plugins/sync-plugin.js
@@ -175,7 +175,10 @@ export const ySyncPlugin = (yXmlFragment, {
       binding.initView(view)
       if (mapping == null) {
         // force rerender to update the bindings mapping
-        binding._forceRerender()
+        eventloop.timeout(0, () => {
+          console.log('force rerender for mapping of new view')
+          binding._forceRerender()
+        })
       }
       onFirstRender()
       return {
